frontend/src/metabase/browse/utils.ts

```diff
@@ -29,25 +31,92 @@ export const groupModels = (
     getCollectionIdForSorting(model.collection),
   );
   const groupsOfModels: SearchResult[][] = Object.values(groupedModels);
-  const sortFunction = (a: SearchResult[], b: SearchResult[]) => {
+  const sortGroupsByCollection = (a: SearchResult[], b: SearchResult[]) => {
     const collection1 = a[0].collection;
     const collection2 = b[0].collection;
 
-    const collection1AL = collection1.authority_level ?? "z";
-    const collection2AL = collection2.authority_level ?? "z";
-    if (collection1AL !== collection2AL) {
-      return collection1AL.localeCompare(collection2AL, locale);
+    // Sort instance analytics collection to the end
+    const collection1IsInstanceAnalyticsCollection =
+      isInstanceAnalyticsCollection(collection1);
+    const collection2IsInstanceAnalyticsCollection =
+      isInstanceAnalyticsCollection(collection2);
+    if (
+      collection1IsInstanceAnalyticsCollection &&
+      !collection2IsInstanceAnalyticsCollection
+    ) {
+      return 1;
+    }
+    if (
+      collection2IsInstanceAnalyticsCollection &&
+      !collection1IsInstanceAnalyticsCollection
+    ) {
+      return -1;
+    }
+
+    const sortValueFromPlugin =
+      PLUGIN_CONTENT_VERIFICATION.sortCollectionsByVerification(
+        collection1,
+        collection2,
+      );
+    if (sortValueFromPlugin) {
+      return sortValueFromPlugin;
     }
 
     const name1 = getCollectionName(collection1);
     const name2 = getCollectionName(collection2);
     return name1.localeCompare(name2, locale);
   };
-  groupsOfModels.sort(sortFunction);
+  groupsOfModels.sort(sortGroupsByCollection);
   return groupsOfModels;
 };
 
 export type BrowseTabId = "models" | "databases";
 
 export const isValidBrowseTab = (value: unknown): value is BrowseTabId =>
   value === "models" || value === "databases";
+
+export type AvailableModelFilters = Record<
+  string,
+  {
+    predicate: (value: SearchResult) => boolean;
+    activeByDefault: boolean;
+  }
+>;
+
+export type ModelFilterControlsProps = {
+  actualModelFilters: ActualModelFilters;
+  handleModelFilterChange: (filterName: string, active: boolean) => void;
+};
+
+export const sortModels = (
+  a: SearchResult,
+  b: SearchResult,
+  localeCode?: string,
+) => {
+  const sortValueFromPlugin =
+    PLUGIN_CONTENT_VERIFICATION.sortModelsByVerification(a, b);
+  if (sortValueFromPlugin) {
+    return sortValueFromPlugin;
+  }
+
+  if (a.name && !b.name) {
+    return -1;
+  }
+  if (!a.name && !b.name) {
+    return 0;
+  }
+  if (!a.name && b.name) {
+    return 1;
+  }
+  if (a.name && !b.name) {
+    return -1;
+  }
+  if (!a.name && !b.name) {
+    return 0;
+  }
+  const nameA = a.name.toLowerCase();
+  const nameB = b.name.toLowerCase();
+  return nameA.localeCompare(nameB, localeCode);
+};
+
+export type ActualModelFilters = Record<string, boolean>;
```

Comment:
@iethree I might be wrong, but I think we might not be able to use the type system to enforce that the keys of this Record match the keys of AvailableModelFilters, since those keys can't be known at compile time - they depend on whether the enterprise plugins are loaded

Reply:
yeah, you're probably right 😞